dashboard/app.py

```diff
@@ -0,0 +1,670 @@
+"""Python Script: Build a dashboard for data visualization"""
+import altair as alt
+from altair.vegalite.v5.api import Chart
+from os import environ, _Environ
+from dotenv import load_dotenv
+from datetime import datetime
+import pandas as pd
+from pandas import DataFrame
+import streamlit as st
+from psycopg2 import connect
+from psycopg2.extensions import connection
+
+
+def get_db_connection(config_file: _Environ) -> connection:
+    """
+    Returns connection to the database
+
+    Args:
+        config (_Environ): A file containing sensitive values
+
+    Returns:
+        connection: A connection to a Postgres database
+    """
+    try:
+        return connect(
+            database=config_file["DB_NAME"],
+            user=config_file["DB_USER"],
+            password=config_file["DB_PASSWORD"],
+            port=config_file["DB_PORT"],
+            host=config_file["DB_HOST"]
+        )
+    except Exception as err:
+        print("Error connecting to database.")
+        raise err
+
+
+def get_database(conn_postgres: connection) -> DataFrame:
+    """
+    Returns redshift database transaction table as a DataFrame Object
+
+    Args:
+        conn_postgres (connection): A connection to a Postgres database
+
+    Returns:
+        DataFrame:  A pandas DataFrame containing all relevant plant data
+    """
+    query = f""
+    df = pd.read_sql_query(query, conn_postgres)
```

Comment:
I was so confused about what was happening here until I remembered the dashboard isn't connected to the db yet, I'm assuming you'll fill in the query once it is